presentation/common/src/presentation-common/RpcRequestsHandler.ts

```diff
@@ -213,4 +202,29 @@ export class RpcRequestsHandler implements IDisposable {
     return this.request<KeySetJSON, SelectionScopeRequestOptions<IModelRpcProps>>(
       this.rpcClient.computeSelection.bind(this.rpcClient), options, ids, scopeId);
   }
+
+  private static removeUnneededPropertiesFromRuleset(ruleset: any) {
+    type RulesetWithRequiredProperties = {
+      [key in keyof Ruleset]-?: true;
+    };
+
+    const rulesetWithRequiredProperties: RulesetWithRequiredProperties = {
+      id: true,
+      rules: true,
+      version: true,
+      requiredSchemas: true,
+      supplementationInfo: true,
+      vars: true,
+    };
```

Comment:
I'd move these two outside the function - no point in creating `rulesetWithRequiredProperties` object every time the function is called. Also, I'd rename the object to `RULESET_SUPPORTED_PROPERIES_OBJ` or something like that.